composeApp/src/main/kotlin/io/github/couchtracker/ui/screens/viewing/DateTimeSection.kt

```diff
@@ -0,0 +1,332 @@
+                    deselect = { sectionState.dateTime = null },
+                )
+            } else {
+                LazyRow(
+                    contentPadding = PaddingValues(horizontal = 16.dp),
+                    horizontalArrangement = Arrangement.spacedBy(8.dp),
+                ) {
+                    items(DateAndTimeSectionChoices.entries) { category ->
+                        FilterChip(
+                            selected = category == dt?.category,
+                            onClick = {
+                                when (category) {
+                                    dt?.category -> {
+                                        sectionState.dateTime = null
+                                    }
+
+                                    DateAndTimeSectionChoices.CUSTOM -> {
+                                        sectionState.customDateDialogVisibility = DATE
+                                    }
+
+                                    else -> {
+                                        sectionState.dateTime = DateAndTimeValue(
+                                            category,
+                                            category.default(Clock.System.now(), TimeZone.currentSystemDefault(), approximateVideoRuntime),
+                                        )
+                                    }
+                                }
+                            },
+                            colors = FilterChipDefaults.filterChipColors(
+                                selectedContainerColor = MaterialTheme.colorScheme.primaryContainer,
+                            ),
+                            label = { Text(category.displayName.string()) },
+                        )
+                    }
+                }
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+private fun CustomDateTimeRow(
+    selectedDateTime: PartialDateTime,
+    setCustomDialogVisibility: (DateTimeSectionState.CustomDateDialogVisibility) -> Unit,
+    deselect: () -> Unit,
+) {
+    val localDate = selectedDateTime.local
+    val dateString = when (localDate) {
+        is PartialDateTime.Local.Year -> localDate.localized(YearSkeleton.NUMERIC).string()
+        is PartialDateTime.Local.YearMonth -> localDate.localized(YearSkeleton.NUMERIC, MonthSkeleton.NUMERIC).string()
+        is PartialDateTime.Local.Date -> localDate.localized(Skeletons.MEDIUM_DATE).string()
+        is PartialDateTime.Local.DateTime -> localDate.localized(Skeletons.MEDIUM_DATE).string()
+    }
+    val timeString = if (localDate is PartialDateTime.Local.DateTime) {
+        localDate.localized(TimeSkeleton.MINUTES).string()
+    } else {
+        null
+    }
+    val tzString = if (selectedDateTime is PartialDateTime.Zoned) {
+        selectedDateTime.zone.id
+    } else {
+        null
+    }
+
+    @Composable
+    fun DateTimePiece(
```

Comment:
Why is this composable inside another composable. From what I can see it's just using `setCustomDialogVisibility` from the main composable, could be easily made top-level private.